### PR TITLE
[SPARK-31503][SQL] fix the SQL string of the TRIM functions

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

override the `sql` method of `StringTrim`, `StringTrimLeft` and `StringTrimRight`, to use the standard SQL syntax.

### Why are the changes needed?

The current implementation is wrong. It gives you a SQL string that returns different result.

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

new tests

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @dongjoon-hyun @HyukjinKwon

**Comment by dongjoon-hyun** [comment]
Thank you for pinging me, @cloud-fan . 
Then, does this affect `branch-2.4`, too?

File: sql/core/src/test/resources/sql-tests/results/string-functions.sql.out [hunk 1]
```diff
@@ -204,65 +204,73 @@ struct<substring(Spark SQL, 5, 1):string>
...
+struct<trim( xyz ):string,ltrim( xyz ):string,rtrim( xyz ):string>
+-- !query output
+xyz	xyz 	 xyz
+
+
 -- !query
 SELECT trim(BOTH 'xyz' FROM 'yxTomxx'), trim('xyz' FROM 'yxTomxx')
 -- !query schema
-struct<trim(yxTomxx, xyz):string,trim(yxTomxx, xyz):string>
+struct<TRIM(BOTH xyz FROM yxTomxx):string,TRIM(BOTH xyz FROM yxTomxx):string>
```
Comment (dongjoon-hyun) [inline sql/core/src/test/resources/sql-tests/results/string-functions.sql.out:218]: Got it.

**Comment by maropu** [comment]
retest this please

**Comment by cloud-fan** [comment]
@dongjoon-hyun I think this bug exists the first day we add `TRIM` or the `sql` method.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @cloud-fan and @maropu .
Merged to master/3.0.

**Comment by dongjoon-hyun** [comment]
Could you make a backporting PR to branch-2.4?

**Comment by dongjoon-hyun** [comment]
cc @holdenk since she is the release manager for 2.4.6.

**Comment by cloud-fan** [comment]
The backport PR: https://github.com/apache/spark/pull/28299

**Pull request closed without merging**